src/server/mod.rs

```diff
@@ -966,14 +980,274 @@ impl LanguageServer for LspServer {
             }
         };
 
-        let items = completion::find_completions(
-            params,
-            contents.as_str(),
-            &sem_pkg,
-        );
+        let walker = flux::ast::walk::Node::Package(&ast_pkg);
+        let mut visitor =
+            crate::visitors::ast::NodeFinderVisitor::new(
+                params.text_document_position.position,
+            );
 
-        let response = lsp::CompletionResponse::List(items);
-        Ok(Some(response))
+        flux::ast::walk::walk(&mut visitor, walker);
+
+        let items = match visitor.node {
+            Some(walk_node) => match walk_node.node {
+                AstNode::CallExpr(call) => {
+                    completion::complete_call_expr(
+                        &params, &sem_pkg, call,
+                    )
+                }
+                AstNode::Identifier(identifier) => {
+                    // XXX: rockstar (6 Jul 2022) - This is helping to complete packages that
+                    // have never been imported. That's probably not a great pattern.
+                    let stdlib_completions: Vec<lsp::CompletionItem> =
+                        if let Some(env) = flux::imports() {
+                            env.iter().filter(|(key, _val)| {
+                            if let Some(package_name) = crate::shared::get_package_name(key) {
+                                fuzzy_match(package_name, &identifier.name)
+                            } else {
+                                false
+                            }
+                        }).map(|(key, _val)| {
+                            #[allow(clippy::unwrap_used)]
```

Comment:
Does this attribute only apply to the next line `let package_name=...` when it's using `unwrap()`? Does the code do the same thing if you declare it at the beginning of the file with `!` in it, like `#![allow(clippy::unwrap_used)]`?